core/src/main/java/com/graphhopper/storage/CHGraphImpl.java

```diff
@@ -261,6 +261,14 @@ public boolean isReadyForContraction() {
         return isReadyForContraction;
     }
 
+    @Override
+    public int getBaseNode(int edge, int adjNode) {
+        EdgeAccess edgeAccess = isShortcut(edge) ? chEdgeAccess : baseGraph.edgeAccess;
+        long edgePointer = edgeAccess.toPointer(edge);
+        int nodeA = edgeAccess.edges.getInt(edgePointer + edgeAccess.E_NODEA);
+        return nodeA != adjNode ? nodeA : edgeAccess.edges.getInt(edgePointer + edgeAccess.E_NODEB);
```

Comment:
With edgeIterableImpl.selectEdgeAccess we would not need to duplicate this here but I see no chance to make this working atm.

Reply:
we could add `EdgeAccess#getOtherNode` ?

Reply:
Yes, this is probably the way to go

Reply:
edgeAccess already has getothernode. done.